src/ipmi.c

```diff
@@ -96,36 +106,81 @@ static void c_ipmi_error(const char *func, int status) {
   ERROR("ipmi plugin: %s failed: %s", func, errbuf);
 } /* void c_ipmi_error */
 
+static void c_ipmi_log(os_handler_t *handler, const char *format,
+                       enum ipmi_log_type_e log_type, va_list ap) {
+  char msg[1024];
+
+  vsnprintf(msg, sizeof(msg), format, ap);
+
+  switch (log_type) {
+  case IPMI_LOG_INFO:
+    INFO("ipmi plugin: %s", msg);
+    break;
+  case IPMI_LOG_WARNING:
+    NOTICE("ipmi plugin: %s", msg);
```

Comment:
Hi, Maryam! 

Thanks for your feedback.

The `OpenIPMI/ipmi_log.h` has the following explanation of error reporting levels:

```
/* General information about something that happens. */
    IPMI_LOG_INFO,

    /* An internal error (not reported directly to the user) occurred,
       but it's not a big deal, the system can still operate. */
    IPMI_LOG_WARNING,

    /* An internal error (not reported directly to the user) occured,
       and it will affect the operation of the system.  This system
       will still operation, but probably in a degraded mode. */
    IPMI_LOG_SEVERE,

    /* An internal error occured, and it is dangerous for the system
       to keep operating. */
    IPMI_LOG_FATAL,

    /* When returning errors to the user, this will often be used to
       report the general cause of the error if the cause would be
       difficult to determine from just the error return code. */
    IPMI_LOG_ERR_INFO,
```
By this explanation, and by typical use of Collectd `WARNING()` (), I think `NOTICE()` suits better.
Of course, I can be wrong...

It would be nice if Collectd will have the similar table of error logging level explanation. (I didn't find it.)

Reply:
Fair enough